scality_sproxyd_client/sproxyd_client.py

```diff
@@ -254,3 +260,57 @@ def handle_200_or_206(response):
         }
 
         return self._do_http('get_object', handlers, 'GET', name, headers)
+
+    def put_object(self, name, body, user_metadata=None):
+        """Connect to sproxyd and put an object."""
+
+        # The Content-Length is automatically set to the correct value.
+        # See `httplib.HTTPConnection.request`
+        headers = {}
+        if user_metadata:
+            headers['x-scal-usermd'] = base64.b64encode(pickle.dumps(user_metadata))
+
+        handlers = {
+            200: lambda _: None,
+        }
+
+        return self._do_http('put_object', handlers, 'PUT', name,
+                             headers=headers, body=body)
+
+    def get_http_conn_for_put(self, name, headers=None):
+        full_url = self.sproxyd_urls.next() + urllib.quote(name)
+        url_pieces = urlparse.urlparse(full_url)
+        conn_pool = self.http_pools.connection_from_host(url_pieces.hostname,
+                                                         url_pieces.port,
+                                                         url_pieces.scheme)
+        timeout_obj = conn_pool.timeout.clone()
+
+        conn = None
+        try:
+            # Request a connection from the queue.
+            conn = conn_pool._get_conn()
+            conn_pool.num_requests += 1
+
+            conn.timeout = timeout_obj.connect_timeout
+
+            conn.putrequest('PUT', url_pieces.path,
+                            skip_host=(headers and 'Host' in headers))
+            if headers:
+                for header, value in headers.iteritems():
+                    conn.putheader(header, str(value))
+            conn.endheaders()
+
+        except (httplib.HTTPException, socket.error) as ex:
+            self.logger.info("Error when trying to PUT %s : %r",
+                             full_url, ex)
+            if conn:
+                # Discard the connection for these exceptions. It will be
+                # be replaced during the next _get_conn() call.
+                conn.close()
+                conn = None
+            # Should we raise an Exception ?
```

Comment:
Yes. Consider what happens if the code falls through here. All of a sudden the method returns `(None, function)`, where the function shouldn't ever be called (it wouldn't work anyway), and the caller should for some reason introduce a `None`-check for the first member.